llama_index/evaluation/correctness.py

```diff
@@ -134,9 +135,20 @@ async def aevaluate(
         )
 
         # Extract from response
-        score_str, reasoning_str = eval_response.split("\n", 1)
+        print(eval_response)
+        # use a regexp to match [SCORE:0.2] in the response
+        match = re.search(r"\[SCORE:(.*?)\]", eval_response)
+        if match:
+            score_str = match.group(1)
+        else:
+            # no score found, maybe retry?
+            pass
```

Comment:
this is fine for now. I think we might need to create another abstraction `InvalidEvaluationResult` that then you can retry post all of the evaluation executions.

Reply:
Other evaluation method ```Relevancy```, ```Faithfulness```, ```Guideline``` do not try to return a confidence score (they do not parse number).  

I think we could update them so they return a confidence score / parsing of YES/NO is done in a more rigid way, but this will be a bigger change. I think  they should have their own PR.

Reply:
ahh right, sorry you're right. We ask there for "YES" or "NO" I believe and then generate a 0/1 score from there.

Reply:
okay this LGTM then to merge :)